pkg/expr/transform.go

```diff
@@ -131,14 +131,17 @@ func QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.Que
 	}
 
 	datasourceID := int64(0)
+	var datasourceUID string
 
 	if req.PluginContext.DataSourceInstanceSettings != nil {
 		datasourceID = req.PluginContext.DataSourceInstanceSettings.ID
+		datasourceUID = req.PluginContext.DataSourceInstanceSettings.UID
 	}
 
 	getDsInfo := &models.GetDataSourceQuery{
 		OrgId: req.PluginContext.OrgID,
 		Id:    datasourceID,
+		Uid:   datasourceUID,
 	}
 
 	if err := bus.Dispatch(getDsInfo); err != nil {
```

Comment:
I think this should perhaps be `ng.DatasourceCache`, looks like this is tech debt for an that results in extra db hits.

I think: if UID != "" ng.DatasourceCache.GetDatasourceByUID else  ng.DatasourceCache.GetDatasourceByID (So uid is preferred, but ID will still work).

Reply:
I'm afraid that requires more refactoring: this function does not know about the cache service.

Reply:
Can consider for future pr